net-libs/libbitcoinconsensus/libbitcoinconcensus-0.15.1.recipe

```diff
@@ -0,0 +1,81 @@
+SUMMARY="Original Bitcoin client"
+DESCRIPTION="BitCoin Core is the backbone of the network. It downloads and, \
+by default, stores the entire history of Bitcoin transactions (which is \
+currently more than 100 GBs); depending on the speed of your computer and \
+network connection, the synchronization process can take anywhere from a few \
+hours to a day or more."
+HOMEPAGE="https://bitcoincore.org"
+COPYRIGHT="2009-2017, Bitcoin Core Developers"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/bitcoin/bitcoin/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="98b3a98a67054123bef6714e4035e18f9250a1b6d63829258ad057906e75d8b1"
+SOURCE_DIR="bitcoin-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libbitcoinconsensus = $portVersion
```

Comment:
libbitconcon**c**ensus?

Reply:
Also in the directory name ...

Reply:
https://github.com/bitcoin/bitcoin/blob/master/Makefile.am#L16 :
```
pkgconfig_DATA = libbitcoinconsensus.pc
```

Reply:
Then the recipe name should be the same :)

Reply:
Good catch! Thanks @Begasus!

Reply:
Still not able to build it on x86_gcc2 (boost_system library not found)

Reply:
I did not try on x86_64. Did you?

Reply:
Nope, not yet

Reply:
--with-boost-libdir="`finddir B_SYSTEM_LIB_DIRECTORY`" (configure option)

Reply:
Also check some missing boost libraries with ./configure --help (still not there though)
> configure: error: No working boost sleep implementation found.

Reply:
I added this to REQUIRES:
```
	lib:libboost_atomic$secondaryArchSuffix
	lib:libboost_chrono$secondaryArchSuffix
	lib:libboost_date_time$secondaryArchSuffix
	lib:libboost_filesystem$secondaryArchSuffix
	lib:libboost_locale$secondaryArchSuffix
	lib:libboost_regex$secondaryArchSuffix
	lib:libboost_system$secondaryArchSuffix
	lib:libboost_thread$secondaryArchSuffix
```
and the same to BUILD_REQUIRES (but with `devel:libboost_` instead of `lib:libboost_`).
I changed the configure call to:
```
	runConfigure ./configure --disable-wallet --with-gui=no \
		--with-boost-libdir="`finddir B_SYSTEM_LIB_DIRECTORY`"
```
Testing this on x86_64 I get:
```
checking for boostlib >= 1.47.0... yes
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system... yes
checking whether the Boost::Filesystem library is available... yes
checking for exit in -lboost_filesystem... yes
checking whether the Boost::Program_Options library is available... yes
checking for exit in -lboost_program_options... yes
checking whether the Boost::Thread library is available... no
checking whether the Boost::Chrono library is available... no
configure: error: hexdump is required for tests
```
I'm not familiar with boost.

Reply:
for hexdump you need to add "--disable-test" at configure

Reply:
https://github.com/Begasus/haikuports/commit/a4529b17f55a2115d7cbe088def2833b091a9c44

This is how far I got so far, still have an error on the sleep implementation ...

Reply:
For the sleep implementation, I found something interesting:
https://github.com/bitcoin/bitcoin/blob/master/configure.ac#L901
Maybe there's an option to disable this somehow?

Reply:
That's strange :0
Even though Boost::Thread and Chrono libraries exist in 
`/boot/system/develop/lib` and `/boot/system/develop/headers/boost`, and we included these in REQUIRES + BUILD_REQUIRES, `configure` says both of these aren't available :(